test/provider/AppDirectory.unittest.ts

```diff
@@ -106,7 +133,7 @@ describe('AppDirectory Unit Tests', () => {
             });
 
             it('Fetches & updates the directory when URLs do not match', async () => {
-                const appDirectory = new AppDirectory();
+                const appDirectory = Injector.get<'APP_DIRECTORY'>(Inject.APP_DIRECTORY);
```

Comment:
Do we definitely want to reuse instances of `AppDirectory` across tests? Particularly since we want to test caching behaviour, we could wind up in a situation where one test depends on a previous test having done *something* first.

Especially feels dangerous since we're directly setting the `_url` inside the `AppDirectory` in some tests. It works out ok because of the line `configStore.config.add({level: 'desktop'}, {applicationDirectory: DEV_APP_DIRECTORY_URL})`, but we're depending on an `onAdd` signal being fired even when the value hasn't changed.

Reply:
Will be fixed with SERVICE-516

Reply:
This is a good point - didn't realise that we were re-using instances between tests.

@ryanpetersonOF I've got some changes that'll rebuild the injector at the start of each test, to prevent any re-use of instances. Let's review tomorrow, and can push up to your fork if all looks good.

Reply:
My laptop is in a slightly dodgy state due to a broken update. Will bump this to SERVICE-516.